week2/homework/src/update.js

```diff
@@ -0,0 +1,23 @@
+const fs = require('fs');
+const list = require('./list');
+const help = require('./help');
+const fileName = './src/data.json';
+
+function update(todoItem, newVal) {
+  return new Promise((resolve, reject) => {
+    list().then(toDos => {
```

Comment:
what if list throws an error?

Reply:
The `list()` handles `errors` itself and the `writeFile` handles it as well, is not that enough?!

Reply:
list() rethrows the error, so it needs to be handled here

Reply:
So I have to use the `.catch` to solve it

Reply:
`.catch(error => console.log(error));`